src/test_typing_extensions.py

```diff
@@ -5390,25 +5416,56 @@ def test_invalid_uses(self):
 
         with self.assertRaisesRegex(
             TypeError,
-            'The last parameter to Concatenate should be a ParamSpec variable',
+            'The last parameter to Concatenate should be a ParamSpec variable or ellipsis',
         ):
             Concatenate[P, T]
 
-        if not TYPING_3_11_0:
-            with self.assertRaisesRegex(
-                TypeError,
-                'each arg must be a type',
-            ):
-                Concatenate[1, P]
+        with self.assertRaisesRegex(
+            TypeError,
+            'is not a generic class',
+        ):
+            Callable[Concatenate[int, ...], Any][Any]
+
+    @skipIf(TYPING_3_11_0, "Args can be non-types in 3.11+")
+    def test_invalid_uses_before_3_11(self):
+        P = ParamSpec('P')
+        with self.assertRaisesRegex(
+            TypeError,
+            'each arg must be a type',
+        ):
+            Concatenate[1, P]
+
+        with self.assertRaisesRegex(
+            TypeError,
+            'each arg must be a type.',
+        ):
+            Concatenate[1, ..., P]
```

Comment:
I wonder if it's really worth adding this test? It seems like in later Python versions we decided that it was better to not raise exceptions in these cases, so that seems like the "desirable" behaviour, since we aim to reflect the behaviour in the latest version of the typing module as much as possible. That means that therefore the behaviour of raising exceptions on these in earlier Python versions is sort-of _undesirable_ behaviour. I wouldn't see it as a big priority for us to fix it, but I'm also not sure it's necessary to have these assertions, if they're arguably asserting undesirable behaviour?

Reply:
Thank you for the reasoning and I do agree with you. The reasons why `_type_check` was made more lax was to prevent bugs and have less restrictions to implement new features.

The only reason I see to keep at least the original test would be to test for a regression like forgetting to call `_type_check` at all.

I would roll back the new test, it does not test something that is desirable to not pass. However, the reason to keep the other one around is not that strong. Would you prefer to keep it? In that case I'll add some comment to it.

Reply:
I've changed the test to assure the `_type_check` but with the *currently* *undesired behavior*.